src/util-file-flash-decompression.c

```diff
@@ -0,0 +1,182 @@
+ * in little-endian order
+ */
+uint32_t FileGetFlashDecompressedLen(const uint8_t *buffer)
+{
+    int a = buffer[4];
+    int b = buffer[5];
+    int c = buffer[6];
+    int d = buffer[7];
+
+    uint32_t value = (((a & 0xff) << 24) | ((b & 0xff) << 16) | ((c & 0xff) << 8) | (d & 0xff));
+
+    return (((value >> 24) & 0x000000FF) | ((value >> 8) & 0x0000FF00) |
+            ((value << 8) & 0x00FF0000) | ((value << 24) & 0xFF000000));
+}
+
+uint8_t FileGetFlashVersion(const uint8_t *buffer, const uint32_t buffer_len)
+{
+    if (buffer_len >= 3)
+        return buffer[3];
+
+    return 0;
+}
+
+/* CWS format */
+/*
+ * | 4 bytes         | 4 bytes    | n bytes         |
+ * | 'CWS' + version | script len | compressed data |
+ */
+int FileDecompressFlashZlibData(DetectEngineThreadCtx *det_ctx,
+                                uint8_t *compressed_data, uint32_t compressed_data_len,
+                                uint8_t *decompressed_data, uint32_t decompressed_data_len)
+{
+    int ret = 1;
+    z_stream infstream;
+    infstream.zalloc = Z_NULL;
+    infstream.zfree = Z_NULL;
+    infstream.opaque = Z_NULL;
+
+    infstream.avail_in = (uInt)compressed_data_len;
+    infstream.next_in = (Bytef *)compressed_data;
+    infstream.avail_out = (uInt)decompressed_data_len;
+    infstream.next_out = (Bytef *)decompressed_data;
+
+    inflateInit(&infstream);
+    int result = inflate(&infstream, Z_NO_FLUSH);
+    switch(result) {
+        case Z_STREAM_END:
+            FileDecompressionSetEvent(det_ctx, FILE_DECODER_EVENT_Z_STREAM_END);
+            break;
+        case Z_OK:
+            FileDecompressionSetEvent(det_ctx, FILE_DECODER_EVENT_Z_OK);
+            break;
+        case Z_DATA_ERROR:
+            FileDecompressionSetEvent(det_ctx, FILE_DECODER_EVENT_Z_DATA_ERROR);
+            ret = 0;
+            break;
+        case Z_STREAM_ERROR:
+            FileDecompressionSetEvent(det_ctx, FILE_DECODER_EVENT_Z_STREAM_ERROR);
+            ret = 0;
+            break;
+        case Z_BUF_ERROR:
+            FileDecompressionSetEvent(det_ctx, FILE_DECODER_EVENT_Z_BUF_ERROR);
+            ret = 0;
+            break;
+        default:
+            FileDecompressionSetEvent(det_ctx, FILE_DECODER_EVENT_Z_UNKNOWN_ERROR);
+            ret = 0;
+            break;
+    }
+    inflateEnd(&infstream);
+
+    return ret;
+}
+
+/* ZWS format */
+/*
+ * | 4 bytes         | 4 bytes    | 4 bytes        | 5 bytes    | n bytes   | 6 bytes         |
+ * | 'ZWS' + version | script len | compressed len | LZMA props | LZMA data | LZMA end marker |
+ */
+int FileDecompressFlashLzmaData(DetectEngineThreadCtx *det_ctx,
+                                uint8_t *compressed_data, uint32_t compressed_data_len,
+                                uint8_t *decompressed_data, uint32_t decompressed_data_len)
+{
+#ifdef HAVE_LIBLZMA
+    int ret = 1;
+    lzma_stream strm = LZMA_STREAM_INIT;
+    lzma_ret result = lzma_alone_decoder(&strm, UINT64_MAX /* memlimit */);
```

Comment:
Is the UINT64_MAX a memory limit here? Might want to choose something sane then.